src/server/models/server-settings/server-settings.ts

```diff
@@ -32,24 +33,27 @@ export interface ServerSettingsValue {
   trustProxy?: TrustProxy;
   strictTransportSecurity?: StrictTransportSecurity;
   auth?: string;
-  settingsUri?: string;
+  settingsLocation?: SettingsLocation;
 }
 
 export type ServerSettingsJS = ServerSettingsValue;
 
-function parseIntFromPossibleString(x: any) {
-  return typeof x === 'string' ? parseInt(x, 10) : x;
-}
-
 function ensureOneOfOrNull<T>(name: string, thing: T, things: T[]): void {
   if (thing == null) return;
   if (things.indexOf(thing) === -1) {
     throw new Error(`'${thing}' is not a valid value for ${name}, must be one of: ${things.join(', ')}`);
   }
 }
 
-var check: Class<ServerSettingsValue, ServerSettingsJS>;
-export class ServerSettings implements Instance<ServerSettingsValue, ServerSettingsJS> {
+function ensureNumber(n: any): void {
```

Comment:
Not blocking but this function's name implies that it can be used to validate any number, however the error that can be thrown explicitly mentions the port.